pkg/asset/installconfig/powervs/client.go

```diff
@@ -1178,3 +1181,103 @@ func (c *Client) CreateSSHKey(ctx context.Context, serviceInstance string, zone
 
 	return nil
 }
+
+// AddIPToLoadBalancerPool adds a server to a load balancer pool for the specified port.
+// @TODO Remove once https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/1679 is fixed.
+func (c *Client) AddIPToLoadBalancerPool(ctx context.Context, lbID string, poolName string, port int64, ip string) error {
+	var (
+		glbOptions    *vpcv1.GetLoadBalancerOptions
+		llbpOptions   *vpcv1.ListLoadBalancerPoolsOptions
+		llbpmOptions  *vpcv1.ListLoadBalancerPoolMembersOptions
+		clbpmOptions  *vpcv1.CreateLoadBalancerPoolMemberOptions
+		lb            *vpcv1.LoadBalancer
+		lbPools       *vpcv1.LoadBalancerPoolCollection
+		lbPool        vpcv1.LoadBalancerPool
+		lbPoolMembers *vpcv1.LoadBalancerPoolMemberCollection
+		lbpmtp        *vpcv1.LoadBalancerPoolMemberTargetPrototypeIP
+		lbpm          *vpcv1.LoadBalancerPoolMember
+		response      *core.DetailedResponse
+		err           error
+	)
+
+	// Make sure the load balancer exists
+	glbOptions = c.vpcAPI.NewGetLoadBalancerOptions(lbID)
+
+	lb, response, err = c.vpcAPI.GetLoadBalancerWithContext(ctx, glbOptions)
+	if err != nil {
+		return fmt.Errorf("failed to get load balancer and the response = %+v, err = %w", response, err)
+	}
+	logrus.Debugf("AddIPToLoadBalancerPool: GLBWC lb = %+v", lb)
+
+	// Query the existing load balancer pools
+	llbpOptions = c.vpcAPI.NewListLoadBalancerPoolsOptions(lbID)
+
+	lbPools, response, err = c.vpcAPI.ListLoadBalancerPoolsWithContext(ctx, llbpOptions)
+	if err != nil {
+		return fmt.Errorf("failed to list the load balancer pools and the response = %+v, err = %w",
+			response,
+			err)
+	}
+
+	// Find the pool with the specified name
+	for _, pool := range lbPools.Pools {
+		logrus.Debugf("AddIPToLoadBalancerPool: pool.ID = %v", *pool.ID)
+		logrus.Debugf("AddIPToLoadBalancerPool: pool.Name = %v", *pool.Name)
+
+		if *pool.Name == poolName {
+			lbPool = pool
+			break
+		}
+	}
+	if lbPool.ID == nil {
+		return fmt.Errorf("could not find loadbalancer pool with name %s", poolName)
+	}
+
+	// Query the load balancer pool members
+	llbpmOptions = c.vpcAPI.NewListLoadBalancerPoolMembersOptions(lbID, *lbPool.ID)
+
+	lbPoolMembers, response, err = c.vpcAPI.ListLoadBalancerPoolMembersWithContext(ctx, llbpmOptions)
+	if err != nil {
+		return fmt.Errorf("failed to list load balancer pool members and the response = %+v, err = %w",
+			response,
+			err)
+	}
+
+	// See if a member already exists with that IP
+	for _, poolMember := range lbPoolMembers.Members {
+		logrus.Debugf("AddIPToLoadBalancerPool: poolMember.ID = %s", *poolMember.ID)
+
+		switch pmt := poolMember.Target.(type) {
+		case *vpcv1.LoadBalancerPoolMemberTarget:
```

Comment:
Is it your intention to do nothing when the `Target` is `LoadBalancerPoolMemberTarget`? If so, it would be good to add a comment explaining why.

Reply:
Thanks for watching out!  Added back!